### ISSUE TITLE
Static Abstract Interface Member issue related to Inheritance and Generic Constraints

### ISSUE THREAD

**Issue by fitdev** [dotnet/roslyn#56753]
**Version Used**: 
DotNet 6 RC1 VS 2022 Preview 4

**Steps to Reproduce**:

```cs

interface IBase { }

abstract class Base : IBase { 

  //Note: Does not define implicit conversion operator!

}

interface IFoo<T> where T : class, IFoo<T> { 
  public static abstract implicit operator T(string value);
}

class Derived : Base, IFoo<Derived> {
  public static implicit operator Derived(string value) { 
    //Do something that returns a new instance of Derived
  }
}

static class Util {

  //Both methods here should be able to work with Derived-s

  static void Method1<T>(string value) where T : Base, IFoo<T> {
    var newT = (T)value; // <-- Error	CS0030: Cannot convert type 'string' to 'T'
  }

  static void Method2<T>(string value) where T : class, IBase, IFoo<T> {
    var newT = (T)value; // <-- No Error
  }

}
```

**Expected Behavior**:
`Method1` should produce no error just as `Method2`, since in both cases `T` is constrained to `IFoo<T>` thereby satisfying the presence of conversion operator.

**Actual Behavior**:
Seemingly when the generic type parameter is constrained to derive from a _particular class_, even though it _has specific interface constraints with static abstracts_, then those static abstract members are not "visible" on `T`. However when `T` is constrained to _interfaces only_, then everything works as expected. But, shouldn't it always work?

**Notes**:
This is a simplified example. In real life, it may not always be possible to constrain `T` to just `IBase` interface as in `Method2`, and a `Base` class constraint may be needed as in `Method1`, so unless the current behavior is fixed, this seriously reduces the usefulness of static abstracts in interfaces.

**Comment by jaredpar** [comment]
@AlekseyTs PTAL

**Comment by AlekseyTs** [comment]
This behavior is expected in the prototype. If there is a specific class constraint, interface constraints do not contribute conversion operators. This should be added to the set of opened design issues.

**Comment by jaredpar** [comment]
@AlekseyTs are we tracking the open issue in a central place? If it's noted in the speclet we can close and reference that.

**Comment by AlekseyTs** [comment]
The current behavior follows the speclet (https://github.com/dotnet/csharplang/blob/main/proposals/static-abstracts-in-interfaces.md#processing-of-user-defined-explicit-conversions). There is also an "Unresolved Questions" section (https://github.com/dotnet/csharplang/blob/main/proposals/static-abstracts-in-interfaces.md#unresolved-questions), which already tracks review/approval of the rules.

**Comment by fitdev** [comment]
@AlekseyTs

Thanks for looking into this. The [doc you referenced](https://github.com/dotnet/csharplang/blob/main/proposals/static-abstracts-in-interfaces.md#unresolved-questions) states that:

> ...the implicit and explicit conversion operators are disallowed in interfaces today. Should they be disallowed as static abstract members as well? Note, the current implementation is adjusted to allow them only in abstract form. If we don't want this behavior after all, there is work to disallow it.

So, at this point is it then not safe at all to define `static abstract [ex|im]plicit operator`s in interfaces, and unlike other types of `static abstract` members (properties, methods, other operators) they may be removed from future previews / versions?

I should also add that the ability to provide DIMs for those, so that they are no longer `abstract` is a super useful thing and hopefully you will keep the conversion operator support as well as allow their implementations to be provided in the interface too.
